e2e/types.go

```diff
@@ -131,7 +133,7 @@ type ScenarioRuntime struct {
 // Config represents the configuration of an AgentBaker E2E scenario.
 type Config struct {
 	// Cluster creates, updates or re-uses an AKS cluster for the scenario
-	Cluster func(ctx context.Context, t *testing.T) (*Cluster, error)
+	Cluster func(ctx context.Context, location string, t *testing.T) (*Cluster, error)
```

Comment:
it might be more future-proof to have Cluster take in a *Scenario instead of only location, assuming we later add something to scenario-specific config that impacts cluster creation